cmds/truncate/truncate.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2013-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Truncate - shrink or extend the size of a file to the specified size
+//
+// Synopsis:
+//     truncate [OPTIONS] [FILE]...
+//
+// Options:
+//     -s: size in bytes
+//     -c: do not create any files
+//
+// Author:
+//     Roland Kammerer <dev.rck@gmail.com>
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+)
+
+var (
+	flags struct {
+		s int64
+		c bool
+	}
+	cmd = "truncate [-c] -s size file..."
+)
+
+func init() {
+	defUsage := flag.Usage
+	flag.Usage = func() {
+		os.Args[0] = cmd
+		defUsage()
+	}
+	flag.BoolVar(&flags.c, "c", false, "Do not create files.")
+	flag.Int64Var(&flags.s, "s", -1, "Size in bytes.")
+	flag.Parse()
+}
+
+func usageAndExit() {
+	flag.Usage()
+	os.Exit(1)
+}
+
+func main() {
+	if flags.s < 0 {
+		log.Println("truncate: ERROR: You have to specify a non-negative size")
+		usageAndExit()
+	}
+
+	if flag.NArg() == 0 {
+		log.Println("truncate: ERROR: You need to specify one or more files.")
+		usageAndExit()
+	}
+
+	for _, fname := range flag.Args() {
+		if _, err := os.Stat(fname); os.IsNotExist(err) && !flags.c {
+			if err := ioutil.WriteFile(fname, []byte{}, 0644); err != nil {
+				log.Fatalf("truncate: ERROR: %v\n", err)
+			}
+		}
+
+		// intentionally ignore, like GNU truncate
+		os.Truncate(fname, flags.s)
```

Comment:
Does GNU truncate check the size is greater or equal to 0?

Reply:
No, a "-3" decreases the current size by 3, a "+3" increases by 3.  0 is the lower limit.  Which will make size a string flag. I'm fine to change that. But where do we stop? I mean all of these commands take units for their size like 'K', 'M'. Would that have to be there too? How "pixel perfect" hast that to be? I mean for example the current "ls" is basically an "ls -a" (prints files starting with a '.'). But does not have an '-a'.